### PR TITLE
Fix: Removing obsolete warning from undesired snapshots

### PR DESCRIPTION
**Pull request by samkevin1**
### Description

This PR removes the `obsolete snapshot` warning from skipped suite tests.

Related to #2442.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [X] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [X] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [X] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/runtime/runners/test.ts [hunk 1]
```diff
@@ -36,6 +36,16 @@ export class VitestTestRunner implements VitestRunner {
   onAfterRunSuite(suite: Suite) {
     if (this.config.logHeapUsage && typeof process !== 'undefined')
       suite.result!.heap = process.memoryUsage().heapUsed
+
+    getTasks(suite).forEach((suiteTask) => {
+      if (suiteTask.mode === 'skip') {
+        getTasks(suiteTask).forEach((task) => {
```
Comment (sheremet-va) [inline packages/vitest/src/runtime/runners/test.ts:42]: I don't think there are any tasks here when suite is skipped. We skip collection phase altogether

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/runtime/runners/test.ts (hunk 1 above)
Comment (samkevin1) [reply packages/vitest/src/runtime/runners/test.ts:42]: Weird, I was logging the tasks and they were there. The warning is also not happening anymore as reported in the issue.

Comment (samkevin1) [reply packages/vitest/src/runtime/runners/test.ts:42]: Yeah, the tasks are there but they have `mode: 'skip'`

Comment (sheremet-va) [reply packages/vitest/src/runtime/runners/test.ts:42]: You can see here that we don't collect tests if suite was marked as skipped, so I'm not sure how you tested:

https://github.com/vitest-dev/vitest/blob/00c432fad6c40b4ff21cc45b18e64f29816062db/packages/runner/src/run.ts#L284

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Can you add more tests? Create a fixture with snapshots and a suite with `.skip`. You can use  `test/test-utils/index.ts` to run Vitest on the fixture.

Then check that the output doesn't contain an error that there are obsolete snapshots

**Comment by samkevin1** [comment]
@sheremet-va Added the tests, not sure why the CI is failing though.

File: test/core/test/snapshot.test.ts [hunk 2]
```diff
@@ -112,3 +115,18 @@ test('renders inline mock snapshot', () => {
...
+
+const root = path.resolve(__dirname, './fixtures')
+function run(testNamePattern?: string) {
+  return runVitest({
+    root,
+    testNamePattern,
+  })
+}
+
+describe('snapshot should not be obsolete', () => {
```
Comment (sheremet-va) [inline test/core/test/snapshot.test.ts:127]: When I run this test without your changes, it gives the same result. Your tests should validate that the change that you made is working.

If you just run this command in main, you can see that there are already no obsolete warnings. Please, create a test that actually tests things.

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Pull request closed without merging**